sdk/cosmosdb/cosmos/src/request/RequestHandler.ts

```diff
@@ -85,11 +97,14 @@ async function httpRequest(
 
   clearTimeout(timeout);
 
-  const result = response.status === 204 || response.status === 304 ? null : await response.json();
+  const result =
+    response.status === 204 || response.status === 304 ? null : JSON.parse(response.bodyAsText);
   const headers = {} as any;
```

Comment:
you could just do

```suggestion
  const headers = response.headers.toJSON();
```

The implementation is basically exactly what you are doing below:

```ts
public toJSON(): RawHttpHeaders {
    const result: RawHttpHeaders = {};
    for (const [key, value] of this._headersMap) {
      result[key] = value;
    }
    return result;
  }
```